### PR TITLE
Improve syntax highlighting

### PR DESCRIPTION
**Pull request by SaraVieira**
- [x] Add padding arround code
- [x] add more themes
- [x] allow user to choose theme

related https://github.com/axodotdev/oranda/issues/24

### CODE REVIEW

**Comment by SaraVieira** [comment]
Would recommend seeing PR like this: https://github.com/axodotdev/oranda/pull/34/files?file-filters%5B%5D=.md&file-filters%5B%5D=.rs&file-filters%5B%5D=.scss&show-viewed-files=true

It hides the PNG's and theme files

File: src/site/markdown/syntax_highlight.rs [hunk 1]
```diff
@@ -1,34 +1,48 @@
...
+    }
+
+    if let Some(syntax_name) = syntax_name {
+        return Ok(syntax_name);
+    }
+
+    Err(OrandaError::Other(
+        "Please add the language to your code snippets".to_owned(),
+    ))
+}
```
Comment (SaraVieira) [inline src/site/markdown/syntax_highlight.rs]: The reason for this function is that `find_syntax_by_extension` will work when your snippet uses `rs` but not when it uses `rust` as the language.

The other one works backwards so trash code it is

File: src/config/mod.rs [hunk 2]
```diff
@@ -1,10 +1,12 @@
 mod oranda;
 mod project;
 pub mod theme;
-
+pub mod types;
```
Comment (ashleygwilliams) [inline src/config/mod.rs]: can we name this differently? maybe themes?

File: src/site/css/stylesheets/_main.scss [hunk 3]
```diff
@@ -169,25 +169,16 @@ h6 {
...
   padding-left: 0;
   margin: 0;
-}
-
-.shiki {
   padding: 1em;
   border-radius: 4px;
 }
 
+
```
Comment (ashleygwilliams) [inline src/site/css/stylesheets/_main.scss]: extraneous space

File: src/site/markdown/mod.rs [hunk 4]
```diff
@@ -10,8 +10,8 @@ use comrak::{self, ComrakOptions, ComrakPlugins};
 mod syntax_highlight;
 use syntax_highlight::syntax_highlight;
 
-pub struct MockAdapter {}
-impl SyntaxHighlighterAdapter for MockAdapter {
+pub struct Adapters {}
```
Comment (ashleygwilliams) [inline src/site/markdown/mod.rs:12]: oh nice

**Review by ashleygwilliams — CHANGES_REQUESTED** [review]
so i think we should nest the syntax highlighting stuff in the markdown dir? since that is where it gets applied? curious what you think. otherwise, there's an extra space that should be removed. this looks good to me once that's handled and we move the theming stuff!

File: src/config/mod.rs (hunk 2 above)
Comment (ashleygwilliams) [reply src/config/mod.rs]: on second though maybe we can nest this in `/markdown/syntax-highlight/` ?

File: src/site/markdown/syntax_highlight.rs (hunk 1 above)
Comment (ashleygwilliams) [reply src/site/markdown/syntax_highlight.rs]: this would be a good code comment here tbh

File: src/site/css/stylesheets/_main.scss (hunk 3 above)
Comment (SaraVieira) [reply src/site/css/stylesheets/_main.scss]: fixed

File: src/config/mod.rs (hunk 2 above)
Comment (SaraVieira) [reply src/config/mod.rs]: I moved a bunch of stuff, let me know what you think

File: src/site/markdown/syntax_highlight.rs (hunk 1 above)
Comment (SaraVieira) [reply src/site/markdown/syntax_highlight.rs]: done!

File: src/site/markdown/syntax_highlight/mod.rs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+use crate::config::Config;
+use crate::errors::*;
+use crate::site::markdown::syntax_highlight::syntax_themes::SyntaxThemes;
+use syntect::highlighting::ThemeSet;
+use syntect::html::highlighted_html_for_string;
+use syntect::parsing::{SyntaxReference, SyntaxSet};
+
+// The reason for this function is that find_syntax_by_extension will work when your snippet uses rs but not when it uses rust as the language.
+
```
Comment (ashleygwilliams) [inline src/site/markdown/syntax_highlight/mod.rs]: you'll want to put a `//` here to show it's a continuous comment

**Review by ashleygwilliams — APPROVED** [review]
very small request re the comment otherwise this is good to go

File: src/site/markdown/syntax_highlight/mod.rs (hunk 5 above)
Comment (SaraVieira) [reply src/site/markdown/syntax_highlight/mod.rs]: Done!

**Pull request merged**